modules/commerce_option_weight/src/Entity/Shipment.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Drupal\commerce_option_weight\Entity;
+
+use Drupal\commerce_option\Entity\ProductOptionValue;
+
+use Drupal\commerce_shipping\Entity\Shipment as CoreShipment;
+
+use Drupal\commerce_order\Entity\OrderItem;
+
+/**
+ * Extends the shipment entity class and modifies the recalculate weight method.
+ *
+ * Adds the option weight value to shipment's total weight.
+ */
+class Shipment extends CoreShipment {
+
+  /**
+   * Recalculates the shipment's weight.
+   */
+  protected function recalculateWeight() {
+    if (!$this->hasItems()) {
+      // Can't calculate the weight if the items are still unavailable.
+      return;
+    }
+
+    /** @var \Drupal\physical\Weight $weight */
+    $weight = NULL;
+
+    foreach ($this->getItems() as $shipment_item) {
+      $shipment_item_weight = $shipment_item->getWeight();
+      $weight = $weight ? $weight->add($shipment_item_weight) : $shipment_item_weight;
+
+      if (!$shipment_item->getOrderItemId()) {
+        continue;
+      }
+
+      $order_item = OrderItem::load(
+        $shipment_item->getOrderItemId()
+      );
+
+      if (!$order_item->hasField('field_options')) {
+        continue;
```

Comment:
Should we be continuing here. Don't we still need to set the weight for other order items that don't have the field opitons?

Reply:
I have actually copied the recalculateWeight method from commerce shipping. 
This is parent recalculateWeight method

`/**
   * Recalculates the shipment's weight.
   */
  protected function recalculateWeight() {
    if (!$this->hasItems()) {
      // Can't calculate the weight if the items are still unavailable.
      return;
    }

    /** @var \Drupal\physical\Weight $weight */
    $weight = NULL;
    foreach ($this->getItems() as $shipment_item) {
      $shipment_item_weight = $shipment_item->getWeight();
      $weight = $weight ? $weight->add($shipment_item_weight) : $shipment_item_weight;
    }
    if ($package_type = $this->getPackageType()) {
      $package_type_weight = $package_type->getWeight();
      $weight = $weight->add($package_type_weight);
    }

    $this->setWeight($weight);
  }`

Inside foreach it only has two lines, so after the two lines I check if it has field_options and if not added continue. So does it makes sense? For getting the orderItem type we have to loop through the shipments. 
Does this makes sense now?

Reply:
Ah, okay. Makes sense Saf. Thanks for clearing that up.